azkaban-common/src/test/java/azkaban/executor/ExecutorManagerTest.java

```diff
@@ -0,0 +1,303 @@
+  }
+
+  /*
+   * Test executor manager active executor reload
+   */
+  @Test
+  public void testSetupExecutorsSucess() throws ExecutorManagerException {
+    Props props = new Props();
+    props.put(ExecutorManager.AZKABAN_USE_MULTIPLE_EXECUTORS, "true");
+    ExecutorLoader loader = new MockExecutorLoader();
+    Executor executor1 = loader.addExecutor("localhost", 12345);
+
+    ExecutorManager manager =
+      new ExecutorManager(props, loader, new HashMap<String, Alerter>());
+    Assert.assertArrayEquals(manager.getAllActiveExecutors().toArray(),
+      new Executor[] { executor1 });
+
+    // mark older executor as inactive
+    executor1.setActive(false);
+    loader.updateExecutor(executor1);
+    Executor executor2 = loader.addExecutor("localhost", 12346);
+    Executor executor3 = loader.addExecutor("localhost", 12347);
+    manager.setupExecutors();
+
+    Assert.assertArrayEquals(manager.getAllActiveExecutors().toArray(),
+      new Executor[] { executor2, executor3 });
+  }
+
+  /*
+   * Test executor manager active executor reload and resulting in no active
+   * executors
+   */
+  @Test
+  public void testSetupExecutorsException() {
+    try {
+      Props props = new Props();
+      props.put(ExecutorManager.AZKABAN_USE_MULTIPLE_EXECUTORS, "true");
+      ExecutorLoader loader = new MockExecutorLoader();
+      Executor executor1 = loader.addExecutor("localhost", 12345);
+
+      ExecutorManager manager =
+        new ExecutorManager(props, loader, new HashMap<String, Alerter>());
+      Set<Executor> activeExecutors =
+        new HashSet(manager.getAllActiveExecutors());
+      Assert.assertArrayEquals(activeExecutors.toArray(),
+        new Executor[] { executor1 });
+
+      // mark older executor as inactive
+      executor1.setActive(false);
+      loader.updateExecutor(executor1);
+      manager.setupExecutors();
+      Assert.fail("Expecting exception, but didn't get one");
+    } catch (ExecutorManagerException ex) {
+      System.out.println("Test true");
+    }
+  }
+
+  /* Test disabling queue process thread to pause dispatching */
+  @Test
+  public void testDisablingQueueProcessThread() throws ExecutorManagerException {
+    ExecutorManager manager = createMultiExecutorManagerInstance();
+    manager.enableQueueProcessorThread();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), true);
+    manager.disableQueueProcessorThread();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), false);
+  }
+
+  /* Test renabling queue process thread to pause restart dispatching */
+  @Test
+  public void testEnablingQueueProcessThread() throws ExecutorManagerException {
+    ExecutorManager manager = createMultiExecutorManagerInstance();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), false);
+    manager.enableQueueProcessorThread();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), true);
+  }
+
+  /* Test submit a non-dispatched flow */
```

Comment:
non-dispatched  = Webserver queued flow. Initially i was planning to use non-dispatched nomenclature.